sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py

```diff
@@ -0,0 +1,541 @@
+                    rospy.sleep(1)
+                    self.test_joint(joint, 'PWM', value=str(float(self.command.value)*(-1.0)),
+                                    prefix=file_prefix + '_minus')
+                    rospy.sleep(1)
+
+    def switch_finger_to_effort(self, finger):
+        joints_to_change = []
+        for joint in self.requested_joints:
+            if finger in joint:
+                joints_to_change.append(joint)
+        temp_controller_helper = ControllerHelper([self._hand_prefix[0] + 'h'], [self._hand_prefix],
+                                                  [joint.lower() for joint in joints_to_change])
+        temp_controller_helper.change_hand_ctrl("effort")
+
+    def switch_finger_to_position(self, finger):
+        joints_to_change = []
+        for joint in self.requested_joints:
+            if finger.lower() in joint.lower():
+                joints_to_change.append(joint)
+        temp_controller_helper = ControllerHelper([self._hand_prefix[0] + 'h'], [self._hand_prefix],
+                                                  [joint.lower() for joint in joints_to_change])
+        temp_controller_helper.change_hand_ctrl("position")
+
+    def switch_to_effort(self):
+        self._controller_helper.change_hand_ctrl("effort")
+        self._mode = 'effort'
+
+    def switch_to_position(self):
+        self._controller_helper.change_hand_ctrl("position")
+        self._mode = 'position'
+
+    def publish_pwm(self, joint, pwm):
+        if 'effort' in self._mode:
+            rospy.loginfo("Applying a PWM of %s to joint %s", str(pwm), joint)
+            self._pwm_command_publishers[joint.upper()].publish(Float64(float(pwm)))
+        else:
+            rospy.logerr("Mode not set to effort, please change this before applying a PWM")
+
+    def test_joint(self, joint, mode='testing', value=0, sleep=3, prefix=''):
+        print joint, joint.lower(), value
+        if '4' in joint and 'th' not in joint.lower():
+            self.free_j4(joint)
+            rospy.sleep(1.5)
+        file_prefix = prefix
+        if prefix == '':
+            file_prefix = raw_input("Name/describe the test (adds text to filename, optional)")
+        if prefix is None:
+            file_prefix = ''
+        rospy.loginfo("Testing joint %s:", joint)
+        self.activate_output(joint, True)
+        if mode == 'testing':
+            self.move_joint_minmax(joint, 'min')
+            self.move_joint_minmax(joint, 'max')
+            self.move_joint_angle(joint, 0)
+        elif mode == 'PWM':
+            rospy.logwarn("val: %s", str(value))
+            self.move_joint_pwm(joint, value)
+            rospy.sleep(sleep)
+        elif mode == 'position':
+            self.move_joint_angle(joint, float(value), wait=True)
+        self.activate_output(joint, False)
+        self.write_output_dictionaries(joint, file_prefix)
+        print
```

Comment:
ieh??

Reply:
it's to try and clear up the console output a bit